src/ipc-wrapper/sub-channel.js

```diff
@@ -0,0 +1,118 @@
+// @ts-check
+import { TypedEmitter } from 'tiny-typed-emitter'
+import { extractMessageEventData } from './utils.js'
+
+export const MANAGER_CHANNEL_ID = '@@manager'
+
+/**
+ * @typedef {Object} Events
+ * @property {(message: any) => void} message
+ */
+
+/**
+ * Node's built-in types for MessagePort are misleading so we opt for this limited type definition
+ * that fits our usage and works in both Node and browser contexts
+ * @typedef {EventTarget & { postMessage: (message: any) => void}} MessagePortLike
+ */
+
+/**
+ * @extends {TypedEmitter<Events>}
+ */
+export class SubChannel extends TypedEmitter {
```

Comment:
wondering if i should avoid using TypedEmitter for this so that this can run in a browser environment as well (it's used for both server and client).

I havea local implementation that works using [EventTarget](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget) based on this helpful [post](https://dev.to/thepassle/events-are-the-shit-b3i). 

one of the main differences is the spec for the [`dispatchEvent()` method](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/dispatchEvent):

> The dispatchEvent() method of the [EventTarget](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget) sends an [Event](https://developer.mozilla.org/en-US/docs/Web/API/Event) to the object, (synchronously) invoking the affected event listeners in the appropriate order. The normal event processing rules (including the capturing and optional bubbling phase) also apply to events dispatched manually with dispatchEvent().

Most notably the part about _synchronously_ invoking listeners. Not sure if the Node implementation adheres to this or if it still behaves similar to an event emitter, but wondering if this specific detail could cause problems in our case or not.

Reply:
i guess the simpler option would be using https://github.com/primus/eventemitter3, which is what rpc-reflector uses.

it's cool that the EventTarget approach works for our use case though!

Reply:
addressed via https://github.com/digidem/mapeo-core-next/pull/261/commits/e067dc37cf03f4b76671324edab73052cf4f5499 (using eventemitter3)